ruddock/modules/budget/helpers.py

```diff
@@ -474,4 +476,35 @@ def validate_payee(payee_id, payee_name):
   # This one's a bit unusual, we want a valid old payee XOR a valid new one
   existing_payee = vu.validate_integer(payee_id, flash_errors=False)
   new_payee = len(payee_name) > 0
-  return (existing_payee != new_payee)
\ No newline at end of file
+  return (existing_payee != new_payee)
+
+
+def download_expenses(filename='expenses.csv'):
+  expenses = get_transactions()
+  ptypes = PaymentType.get_all()
+  fields = ["expense_id", "budget_name", "fyear_num", "date_incurred",
+            "description", "cost", "payment_id", "payee_name", "payment_type",
+            "account_name", "check_no"]
+  # If the first title is "ID" there is a formatting issue with Excel
+  titles = ["Expense ID", "Budget", "FY", "Date Incurred", "Description",
+            "Amount", "Payment ID", "Payee", "Type", "Account", "Check"]
+
+  # The approach is outlined at "Python Flask send_file StringIO blank files"
```

Comment:
Probably better to link the SO questions directly. If you hit "share" by the answer, it'll give you a short link of the form `https://stackoverflow.com/a/########`